dfu/boot: request upgrade & confirm for multimage

nvlsianpu opened the pull request:
Add boot_write_confirm_multi() and boot_request_upgrade_multi() APIs so
that the user can set the image with given index as pending, confirmed.

This is needed for enabling the mcuboot multi-mage boot feature by
the zephyr.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu committed:
dfu/boot: request upgrade & confirm for multimage

Add boot_write_confirm_multi() and boot_request_upgrade_multi() APIs so
that the user can set the image with given index as pending, confirmed.

This is needed for enabling the mcuboot multi-mage boot feature by
the zephyr.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu commented:
@de-nordic This is simple enabling what bootutil_public already provide.

As additional patch.
I'm also thinkin that would be good to make boot_request_upgrade() and boot_write_img_confirmed() aware on to which pair running image belongs, so they will operate on slot from which it were called or appropriate image pair.
This will need to use more sophisticated flash_area  dispatching you are working around currently.

sigvartmh reviewed (approved):
Looks reasonable to me.

de-nordic reviewed (approved):
Looks OK.

nvlsianpu committed:
dfu/boot: add CONFIG_UPDATEABLE_IMAGE_NUMBER option

This option is deciding whether bootutil_public library supports
multi-image procedures. MCUboot have its own definition which is
unavailable zephyr application. Introduced option declares similar
option for the application.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu committed:
tests/subsys/dfu: add testsuite for multi-image support

Added tests for exercise boot_request_upgrade_multi() API.
Added tests for exercise boot_write_img_confirmed_multi() API.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>
